src/memray/_memray/tracking_api.cpp

```diff
@@ -197,6 +252,7 @@ PythonStackTracker::emitPendingPushes()
 inline int
 PythonStackTracker::getCurrentPythonLineNumber()
 {
+    reloadStackIfTrackerChanged();
```

Comment:
This is called  twice in the code path from `pushPythonFrame` and maybe others. Maybe we should move this to the callers (`pushPythonFrame` and `trackAllocationImpl`). Seems that that way we may also cover other stuff because `trackAllocationImpl` also calls `emitPendingPops` that also calls `reloadStackIfTrackerChanged` so we can eliminate that call as well.